src/widgets/vitals/vitals-card.resource.tsx

```diff
@@ -129,52 +131,60 @@ export function savePatientVitals(
     },
     signal: abortController.signal,
     body: {
-      encounterDatetime: encounterDatetime,
-      encounterProviders: [
-        {
-          provider: "f9badd80-ab76-11e2-9e96-0800200c9a66",
-          encounterRole: "240b26f9-dd88-4172-823d-4a8bfeb7841f"
-        }
-      ],
-      location: "b1a8b05e-3542-4037-bbd3-998ee9c40574",
       patient: patientUuid,
-      encounterType: "67a71486-1a54-468f-ac3e-7091a9a79584",
-      form: "a000cb34-9ec1-4344-a1c8-f692232f6edd",
-      obs: [
-        {
-          concept: SYSTOLIC_BLOOD_PRESSURE_CONCEPT,
-          value: vitals.systolicBloodPressure
-        },
-        {
-          concept: DIASTOLIC_BLOOD_PRESSURE_CONCEPT,
-          value: vitals.diastolicBloodPressure
-        },
-        {
-          concept: PULSE_CONCEPT,
-          value: vitals.heartRate
-        },
-        {
-          concept: OXYGENATION_CONCEPT,
-          value: vitals.oxygenSaturation
-        },
-        {
-          concept: TEMPERATURE_CONCEPT,
-          value: vitals.temperature
-        },
-        {
-          concept: WEIGHT_CONCEPT,
-          value: vitals.weight
-        },
-        {
-          concept: HEIGHT_CONCEPT,
-          value: vitals.height
-        }
-      ],
-      orders: []
+      encounterDatetime: encounterDatetime,
+      location: location,
+      encounterType: VITALS_ENCOUNTER_TYPE,
+      form: VITALS_FORM,
+      obs: isVitalValid(vitals)
     }
   });
 }
 
+function isVitalValid(vitals: Vitals): any[] {
+  return Object.entries(vitals)
+    .filter(el => el[1] != null)
+    .map(validVitals => {
+      switch (validVitals[0]) {
```

Comment:
This switch case doesn't have a default case. It will return `undefine` from the map function if  `validVitals[0]` doesn't match any case. Isn't it? 🤔

Reply:
yeap, resolved it